components/CreateSetupLink.tsx

```diff
@@ -0,0 +1,242 @@
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        tenant,
+        product,
+        type,
+      }),
+    });
+    if (res.ok) {
+      setLoading(false);
+      const json = await res.json();
+      setUrl(json.data.url);
+      setStatus({ status: 'SUCCESS', linkPanelStatus: 'UNKNOWN' });
+      setTimeout(() => setStatus({ status: 'UNKNOWN', linkPanelStatus: 'UNKNOWN' }), 2000);
+    } else {
+      // save failed
+      setLoading(false);
+      setStatus({ status: 'ERROR', linkPanelStatus: 'UNKNOWN' });
+      setTimeout(() => setStatus({ status: 'UNKNOWN', linkPanelStatus: 'UNKNOWN' }), 2000);
+    }
+  };
+  const regenerateLink = async (event) => {
+    event.preventDefault();
+    setLoading1(true);
+    setDelModalVisible(!delModalVisible);
+    const { tenant, product, type } = formObj;
+
+    const res = await fetch('/api/admin/setup-links', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        tenant,
+        product,
+        type,
+        regenerate: true,
+      }),
+    });
+    if (res.ok) {
+      setLoading1(false);
+      const json = await res.json();
+      setUrl(json.data.url);
+      setStatus({ linkPanelStatus: 'SUCCESS', status: 'UNKNOWN' });
+      setTimeout(() => setStatus({ linkPanelStatus: 'UNKNOWN', status: 'UNKNOWN' }), 2000);
+    } else {
+      // save failed
+      setLoading1(false);
+      setStatus({ linkPanelStatus: 'ERROR', status: 'UNKNOWN' });
+      setTimeout(() => setStatus({ linkPanelStatus: 'UNKNOWN', status: 'UNKNOWN' }), 2000);
+    }
+  };
+  const copyUrl = () => {
+    navigator.clipboard.writeText(url);
+    setStatus({ linkPanelStatus: 'COPIED', status: 'UNKNOWN' });
+    setTimeout(() => setStatus({ linkPanelStatus: 'UNKNOWN', status: 'UNKNOWN' }), 2000);
+  };
+  const [{ status, linkPanelStatus }, setStatus] = useState<{
+    status: 'UNKNOWN' | 'SUCCESS' | 'ERROR' | 'COPIED';
+    linkPanelStatus: 'UNKNOWN' | 'SUCCESS' | 'ERROR' | 'COPIED';
+  }>({
+    status: 'UNKNOWN',
+    linkPanelStatus: 'UNKNOWN',
+  });
+  const getHandleChange = (event: FormEvent) => {
+    const target = event.target as HTMLInputElement | HTMLTextAreaElement;
+    setFormObj((cur) => ({ ...cur, [target.name]: target.value }));
+  };
+  const [url, setUrl] = useState<string>('');
+  const [loading, setLoading] = useState<boolean>(false);
+  const [loading1, setLoading1] = useState<boolean>(false);
+  const [formObj, setFormObj] = useState<Record<string, string>>({
+    tenant: '',
+    product: '',
+    type: props.service || 'sso',
+  });
+  const [delModalVisible, setDelModalVisible] = useState(false);
+  const toggleDelConfirm = () => setDelModalVisible(!delModalVisible);
+
+  return (
+    <>
+      <Link href='/admin/sso-connection' className='btn-outline btn items-center space-x-2'>
```

Comment:
The back link here points to sso connection page ...

Reply:
Otherwise LGTM.

Reply:
Good spot, cc: @ukrocks007